### ISSUE TITLE
Actions require an explicit HttpMethod when mixing Routes and HttpMethod attributes

### ISSUE THREAD

**Issue by chrisbecke** [domaindrivendev/Swashbuckle.AspNetCore#1059]
SwaggerGen emits this error when route parameters are split between a `Route` and a `HttpMethod` attribute on a method:

    System.NotSupportedException: Ambiguous HTTP method for action - webapi.Controllers.UserController.DeleteUser (webapi). Actions require an explicit HttpMethod binding for Swagger 2.0
       at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreatePathItem(IEnumerable`1 apiDescriptions, ISchemaRegistry schemaRegistry)
       at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
       at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath, String[] schemes)
       at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
       at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

### Versions

* Microsoft.AspNetCore 2.2.0
* Swashbuckle.AspNetCore 4.0.1

### To reproduce

A new project was created using the cli:

    dotnet new webapi
    dotnet add package Swashbuckle.AspNetCore;

with the usual boilerplate added to `Startup.cs`.

This controller is added:

``` csharp
    [Route("api")]
    public class UserController : ControllerBase
    {
        [Route("users")]
        [HttpGet]
        public IEnumerable<string> Get()
        {
            return new string[] { "value1", "value2" };
        }

        [Route("libraries/{library}/user")]
        [HttpGet("{id}")]
        public string GetUser(string library, int id)
        {
            return "value";
        }

/* working form
        [Route("libraries/{library}/user/{id}")]
        [HttpGet()]
        public string GetUser(string library, int id)
        {
            return "value";
        }
*/
    }
}
```

There are two versions ot the `GetUser` method. The first version, where the parameters are split between the Route and the Controller, causes `SwaggerGen` to emit an error. The 2nd phrasing where all the parameters are in the `Route` does not.

I would expect `SwaggerGen` to treat both versions the same way.

**Comment by domaindrivendev** [comment]
Independently of Swashbuckle, I think you're misunderstanding the semantics of the `Route` and `HttpGet` attributes. What you're actually doing in the `GetUser` example above is creating two separate URL's for accessing the user information:

```
/api/libraries/{library}/user
/api/{id}
```

If you don't believe me, try the following curl commands and you'll see that they both return the string "value":

```
curl http://localhost/api/libraries/foo/user
curl http://localhost/api/123
```

In fact, you could call the first endpoint with any HTTP method (GET, POST, DELETE etc.) and it will work because that route was defined without the `Route` attribute.

Long story short, you _can_ combine the attributes if, and only if, you DO NOT specify a path with the `HttpGet` attribute. Once you include a path, you're essentially defining a separate endpoint for the operation. If there's no good reason to use separate attributes, I would recommend combining everything into the `HttpGet` attribute as this is the most succinct and explicit approach. For example:

```csharp
[HttpGet("libraries/{library}/user/{id}")]
public string GetUser(string library, int id)
{
     return "value";
}
```

**Comment by chrisbecke** [comment]
At this time, I am trying to figure out why I was not just combining everything into the HttpGet.

I surely knew, prior to your final example, that HttpGet() can take a route.
